components/Comment/CommentEditor.tsx

```diff
@@ -0,0 +1,133 @@
+import { useEffect, useRef, useState } from "react";
+import dynamic from "next/dynamic";
+import CreateBountyBtn from "~/components/Bounty/CreateBountyBtn";
+import Button from "~/components/Form/Button";
+import { css, StyleSheet } from "aphrodite";
+import isQuillEmpty from "./lib/isQuillEmpty";
+
+const ReactQuill = dynamic(() => import("react-quill"), { ssr: false });
+
+type CommentEditorArgs = {
+  previewWhenInactive?: boolean;
+  placeholder?: string;
+  handleSubmit: Function;
+  content?: string;
+  allowBounty?: boolean;
+};
+
+const CommentEditor = ({
+  previewWhenInactive = false,
+  placeholder = "Add comment or start a bounty",
+  handleSubmit,
+  content = "",
+  allowBounty = false,
+}: CommentEditorArgs) => {
+  const editorRef = useRef<any>(null);
+  const [isSubmitDisabled, setIsSubmitDisabled] = useState(true);
+  const [_previewWhenInactive, _setPreviewWhenInactive] =
+    useState(previewWhenInactive);
+  const [_isPreview, _setIsPreview] = useState(previewWhenInactive);
+  const _isPreviewRef = useRef(_isPreview);
+  const [_content, _setContent] = useState<string>(content);
+  const [_isFocused, _setIsFocused] = useState(false);
+
+  const handleEditorChange = (value, delta, source, editor) => {
+    const editorContents = editor.getContents();
+    if (isQuillEmpty(editorContents)) {
+      setIsSubmitDisabled(true);
+    } else {
+      setIsSubmitDisabled(false);
+    }
+
+    _setContent(value);
+  };
+
+  useEffect(() => {
+    const _handleClick = (e) => {
```

Comment:
This whole section may be needed -- without diving into the logic a bit more it's hard for me to tell but on initial glance it looks kind of opaque and may be hard to maintain.

Just marking as a section that could be bug prone

Reply:
Yeah I agree, I kinda don't like it. 

I  consolidated this in the other PR. LMK if you have thoughts about how to make it even cleaner https://github.com/ResearchHub/researchhub-web/blob/81e4a51a13dc97e960418cabb55b210121c6211d/components/Comment/CommentEditor.tsx#L48